packages/tradershub/src/flows/FirstSignup/CitizenshipModal/CitizenshipModal.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from 'react';
+import { useFormikContext } from 'formik';
+import { Modal } from '@/components';
+import { useClientCountry, useResidenceList } from '@deriv/api';
+import { LabelPairedChevronDownMdRegularIcon } from '@deriv/quill-icons';
+import { Button, Checkbox, Dropdown, Text } from '@deriv-com/ui';
+import { TFirstSignupFormValues } from '../FirstSignupWrapper/FirstSignupWrapper';
+
+type TCitizenshipModal = {
+    onClickNext: VoidFunction;
+};
+
+const CitizenshipModal = ({ onClickNext }: TCitizenshipModal) => {
+    const { data: residenceList, isLoading: residenceListLoading } = useResidenceList();
+    const { data: clientCountry, isLoading: clientCountryLoading } = useClientCountry();
+    const [isCheckBoxChecked, setIsCheckBoxChecked] = useState(false);
+    const { values, setFieldValue } = useFormikContext<TFirstSignupFormValues>();
+    const isBrazil = values.country === 'br';
+
+    useEffect(() => {
+        if (residenceList?.length && clientCountry && values.country === '') {
+            setFieldValue('country', clientCountry);
+        }
+    }, [clientCountry, setFieldValue, residenceList, values.country]);
+
+    // Add <Loading /> here later when it's created
+    if (clientCountryLoading && residenceListLoading) return null;
+
+    return (
+        <Modal className='h-full rounded-default max-w-[328px] md:max-w-[440px]'>
```

Comment:
@sergei-deriv let's use ReactModal instead of Modal.

Reply:
@mahdiyeh-deriv honestly, I don't agree with this approach))) but if everybody in the team wants to use this library then ok, I'll modify it